lib/rubocop/plugin/plugin_constantizer.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require_relative '../feature_loader'
+require_relative 'plugin_load_error'
+
+module RuboCop
+  module Plugin
+    # Determines and loads the constantized class for a given plugin.
+    # Handles plugin class resolution from require paths or gemspec metadata.
+    # @api private
+    class PluginConstantizer
```

Comment:
This class also looks a bit weird to me. At the very least I think we can come up with a more descriptive name here or move the logic to the plugin loader class.

Reply:
This has been inlined into `RuboCop::Plugin::Loader`.